sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -680,14 +682,22 @@ void gnrc_rpl_recv_DIO(gnrc_rpl_dio_t *dio, kernel_pid_t iface, ipv6_addr_t *src
     /* gnrc_rpl_parent_add_by_addr should have set this already */
     assert(parent != NULL);
 
-    parent->rank = byteorder_ntohs(dio->rank);
+    uint16_t dio_rank = byteorder_ntohs(dio->rank);
```

Comment:
this change looks like an unrelated optimisation? I also doubt, that it would be wise to remove a parent just because its rank changed.